standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -379,6 +379,102 @@ public void markCompacted(CompactionInfo info) throws MetaException {
     }
   }
 
+  /**
+   * Mark the cleaning start time for a particular compaction
+   *
+   * @param info info on the compaction entry
+   */
+  @Override
+  @RetrySemantics.CannotRetry
```

Comment:
I'm not 100% on these, but shouldn't this have the @RetrySemantics.ReadOnly annotation?